### PR TITLE
Fix list aggregate events performance issues

### PR DESCRIPTION
**Pull request by stefanvozd**
Fixes issues: 
- parallel segment reading
- initial sequence number always starts from zero
- events prefetching 
- reading events on gRPC threads
- reading snapshots on gRPC threads

### CODE REVIEW

File: axonserver/src/main/java/io/axoniq/axonserver/localstorage/AggregateReader.java [hunk 1]
```diff
@@ -85,28 +85,30 @@ public long readHighestSequenceNr(String aggregateId, int maxSegmentsHint, long
...
+    }
+
+
+    private Mono<SerializedEvent> snapshot(String aggregateId,
+                                   boolean useSnapshots,
+                                   long minSequenceNumber,
+                                   long maxSequenceNumber){
+        if (useSnapshots){
+            return snapshotReader.snapshot(aggregateId, minSequenceNumber, maxSequenceNumber)
+            .filter(snapshot -> snapshot.getAggregateSequenceNumber() >= maxSequenceNumber);
```
Comment (abuijze) [inline axonserver/src/main/java/io/axoniq/axonserver/localstorage/AggregateReader.java:108]: I don't understand this one. Should it be >= minSequenceNumber instead?

File: axonserver/src/main/java/io/axoniq/axonserver/localstorage/DataFeatcherSchedulerProvider.java [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+import java.util.concurrent.Executors;
+import java.util.function.Supplier;
+
+/**
+ * @author Stefan Dragisic
+ * @author Sara Pellegrini
+ * @since 4.5.3
+ */
+
+public class DataFeatcherSchedulerProvider implements Supplier<ExecutorService> {
```
Comment (abuijze) [inline axonserver/src/main/java/io/axoniq/axonserver/localstorage/DataFeatcherSchedulerProvider.java:15]: Typo in class name. 

This class is fine for an urgent bug-fix, but we should really get rid of 'global variables' as soon as we can.

File: axonserver/src/main/java/io/axoniq/axonserver/localstorage/LocalEventStore.java [hunk 3]
```diff
@@ -87,7 +86,7 @@
     private final Logger logger = LoggerFactory.getLogger(LocalEventStore.class);
     private final Map<String, Workers> workersMap = new ConcurrentHashMap<>();
     private final EventStoreFactory eventStoreFactory;
-    private final ExecutorService dataFetcher;
+    public final ExecutorService dataFetcher;
```
Comment (abuijze) [inline axonserver/src/main/java/io/axoniq/axonserver/localstorage/LocalEventStore.java:89]: Let's avoid public fields

File: axonserver/src/main/java/io/axoniq/axonserver/message/event/EventDispatcher.java [hunk 4]
```diff
@@ -122,7 +122,7 @@ public EventDispatcher(EventStoreLocator eventStoreLocator,
                            GrpcFlowControlExecutorProvider grpcFlowControlExecutorProvider,
                            @Value("${axoniq.axonserver.event.aggregate.retry.attempts:3}") int maxRetryAttempts,
                            @Value("${axoniq.axonserver.event.aggregate.retry.delay:100}") long retryDelayMillis,
-                           @Value("${axoniq.axonserver.event.aggregate.prefetch:100}") int aggregateEventsPrefetch) {
+                           @Value("${axoniq.axonserver.event.aggregate.prefetch:5}") int aggregateEventsPrefetch) {
```
Comment (abuijze) [inline axonserver/src/main/java/io/axoniq/axonserver/message/event/EventDispatcher.java:125]: I think we document the default values in the ref guide. If we change it, we should document the new default there

**Review by abuijze — CHANGES_REQUESTED** [review]

**Review by MGathier — APPROVED** [review]

**Review by saratry — APPROVED** [review]

**Pull request merged**